### PR TITLE
Add getItemsByGroup function

### LINKED ISSUE #124: Allow devs to select config items by group
```js
[
  {
    "type": "input",
    "messageKey": "COOL_THINGS[0]",
    "defaultValue": "",
    "label": "Input Field",
    "group": "COOL_THINGS"
  },
  {
    "type": "input",
    "messageKey": "COOL_THINGS[1]",
    "defaultValue": "",
    "label": "Input Field",
    "group": "COOL_THINGS"
  },
]
```
Then to get the items in the group you would do:

```js
var coolThingsItems = clayConfig.getItemsByGroup('COOL_THINGS')
```

As a temporary workaround you can do the following in your custom function:

```js
clayConfig.getItemsByGroup = function(group) {
  return this.getAllItems().filter(function(item) {
    return item.config.group === group;
  });
}

var coolThingsItems = getItemsByGroup('COOL_THINGS')
```

### PR DESCRIPTION
**Pull request by gregoiresage**
Resolves #124

### CODE REVIEW

**Comment by gregoiresage** [comment]
Still need to update completely the README :-p

File: test/spec/lib/clay-config.js [hunk 1]
```diff
@@ -287,6 +289,20 @@ describe('ClayConfig', function() {
...
+    it('it returns the correct items', function() {
+      var config = fixtures.config([
+        {type: 'input', clayId: 0, group: 'group1'},
+        {type: 'text', clayId: 1, group: 'group2'},
+        {type: 'input', clayId: 2, group: 'group1'}
+      ]);
+      var clayConfig = fixtures.clayConfig(config);
+      assert.deepEqual(clayConfig.getItemsByGroup('group1')[0].config, config[0]);
+      assert.deepEqual(clayConfig.getItemsByGroup('group1')[1].config, config[2]);
+      assert.deepEqual(clayConfig.getItemsByGroup('group2')[0].config, config[1]);
```
Comment (keegan-lillo) [inline test/spec/lib/clay-config.js]: Rather do:

```js
var config = fixtures.config([
  {type: 'input', id: 'g1-0', group: 'group1'},
  {type: 'text', id: 'g2-0', group: 'group2'},
  {type: 'input', id: 'g1-1', group: 'group1'}
]);

var clayConfig = fixtures.clayConfig(config);
assert.deepEqual(clayConfig.getItemsByGroup('group1'), [
  clayConfig.getItemById('g1-0'),
  clayConfig.getItemById('g1-1')
]);
assert.deepEqual(clayConfig.getItemsByGroup('group2'), [
  clayConfig.getItemById('g1-1')
]);
```
This also ensure the result *only* contains the items in that group

**Comment by keegan-lillo** [comment]
please update from master then will merge :)

**Comment by keegan-lillo** [comment]
Can you also please add the `group` property for each component in the docs? eg: https://github.com/gregoiresage/clay/blob/%23124/getItemsByGroup/README.md#input

**Comment by gregoiresage** [comment]
group	string	The optional group of the item

Something like that ? Tell me what do you want in the description

**Comment by keegan-lillo** [comment]
Something like:

Set this to allow this item, along with other items sharing the same group to be looked up using Clay.getItemsByGroup() in your custom function. (link to custom function doc)

**Comment by gregoiresage** [comment]
ok

**Comment by keegan-lillo** [comment]
@gregoiresage You missed the submit component ;)

**Comment by gregoiresage** [comment]
The submit button has no id, is this normal ?

**Comment by keegan-lillo** [comment]
Oh whoops, and neither does the button component either. Yeh both of those should be updated.

**Comment by keegan-lillo** [comment]
They should both have message keys too

**Comment by keegan-lillo** [comment]
Finished it off in #147 Thanks for the PR

**Pull request closed without merging**